### ISSUE TITLE
About official multi-platform initial support for Arm64

### ISSUE THREAD

**Issue by lubinsz** [kubernetes/kubernetes#86295]
Hi,
Currently, I checked the design-proposal of multi-platform (https://github.com/kubernetes/community/blob/a8944bea54d4bb854bdd5da98ecae3ebc7a18d9a/contributors/design-proposals/multi-platform.md),  and in this document it says so: "When all e2e tests are passing for a given platform; the platform should be officially supported by the Kubernetes team. At the time of writing, amd64 is in the officially supported category. "

So my question is, we can already pass the conformance test on Arm64 and we can guarantee that the test runs with the latest development version normally every day: https://k8s-testgrid.appspot.com/sig-node-arm64#conformance
Also for other e2e tests, the result is same with amd64.

So, what else do I need to do in order for the community to officially announce support for arm64?

Based on the suggestions from @dims , I listed some tasks roughly:
* [x] publish the daily test results for the conformance test on Arm64
       https://k8s-testgrid.appspot.com/sig-node-arm64#conformance
* [x] add Arm64 support to pull-kubernetes-integration 
* [x] publish the daily test results for the equivalent of pull-kubernetes-integration on Arm64
        https://k8s-testgrid.appspot.com/sig-node-arm64#integration-tests
* [x] add Arm64 support to pull-kubernetes-e2e
* [x] publish the daily test results for the equivalent of pull-kubernetes-e2e on Arm64
* [x] add Arm64 support to pull-kubernetes-node-e2e
* [x] publish the daily test results for the equivalent of pull-kubernetes-node-e2e on Arm64
       https://k8s-testgrid.appspot.com/sig-node-arm64#nodeconformance
* [x] add Arm64 support to k8s/kind
* [x] publish the daily test results for the equivalent of pull-kubernetes-e2e-kind on Arm64
        https://k8s-testgrid.appspot.com/sig-node-arm64#kind,%20master%20(dev,%20ARM64)
* [x] add Arm64 support to kubeadm
* [x] support some of the installers? (updated documentation?): private installer.
* [x] update documentation about 'Installing kubeadm' on Arm64 https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/
* [x] update documentation about 'Creating a single control-plane cluster with kubeadm' on Arm64
https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/
* [x] fully transparent build scripts. 
...
Basic CNI-s support:
* [x] Flannel
* [x] publish the daily test results for the conformance test on Arm64 with Flannel
* [x] Weave
* [x] publish the daily test results for the conformance test on Arm64 with Weave
* [x] Calico
* [x] publish the daily test results for the conformance test on Arm64 with Calico

Extended CNI-s support: https://github.com/kubernetes/kubernetes/issues/87876

CRI-s support:
* [x] containerd
* [x] publish the daily test results for the conformance test on Arm64 with containerd
       https://k8s-testgrid.appspot.com/conformance-arm#Periodic%20Arm64%20conformance%20test%20with%20containerd

Cloud-provider:
* [x] publish the daily test results for the conformance test on AWS Arm64 instances
       https://k8s-testgrid.appspot.com/conformance-arm#Periodic%20Arm64%20conformance%20test%20on%20AWS
* [x] publish the daily test results for the conformance test on Packet.net Arm64 instances
Thanks.

**Comment by lubinsz** [comment]
Hi @luxas 
Any comments?

**Comment by lubinsz** [comment]
cc @zhlhahaha @chendave

**Comment by dims** [comment]
@lubinsz Here's my thought / brain dump
- We should publish the daily test results for the equivalent of `pull-kubernetes-e2e-gce` job, NOT just conformance
- Do we support the basic tools used the community (`kind`, `kubeadm`) and ensure their documentation is updated to reflect.
- Do we support some of the installers? (updated documentation?)
- Do we have fully transparent build scripts? (To ensure that folks can recreate the CI jobs)
- Are the job running in publicly available environments (cloud providers? CI?) (To ensure that folks can validate the success seen in CI)
- What options are available for some of the default addons (List of supported CNIs for example? do we have documentation?)
- Do we know what does not work? Do we have tracking issues in different repositories with some umbrella issue somewhere?
- Where do folks ask for support if they run into issues? (Who is going to be present to help with issues, debug/recreate problems?)

That's my thinking for the moment, we need to turn this into a better policy (as opposed to the single line in the design proposal)

**Comment by lubinszARM** [comment]
@dims 
I see. Thank you very much! :)

**Comment by lubinsz** [comment]
cc @BenTheElder

**Comment by lubinsz** [comment]
Hi @dims 
> We should publish the daily test results for the equivalent of pull-kubernetes-e2e-gce job, NOT just conformance
> 
How to publish the daily test results for the equivalent of pull-kubernetes-e2e-gce job?
We have tested it in our internal Arm64 cluster, and the result seems OK.

>Do we support the basic tools used the community (kind, kubeadm) and ensure their documentation is updated to reflect.
>
kubeadm is OK. And @zhlhahaha is enabling kind right now.

>Do we support some of the installers? (updated documentation?)
> ....
> 
Ongoing.

**Comment by dims** [comment]
@lubinsz https://github.com/kubernetes/test-infra/blob/c5595623a73069fd71f977f097ae0372b7916879/testgrid/conformance/README.md#display-conformance-tests-with-testgrid on how to publish results to testgrid ( cc @BenTheElder )

**Comment by dims** [comment]
@lubinsz also see the notes in thread https://groups.google.com/d/msg/kubernetes-sig-testing/WULGoGS1rS4/5mn09AaNAwAJ

**Comment by lubinsz** [comment]
> @lubinsz https://github.com/kubernetes/test-infra/blob/c5595623a73069fd71f977f097ae0372b7916879/testgrid/conformance/README.md#display-conformance-tests-with-testgrid on how to publish results to testgrid ( cc @BenTheElder )

I see. The same method like publishing conformance test results into testgrid.
Thank you very much!

**Comment by lubinsz** [comment]
cc @Jingzhao123

**Comment by lubinszARM** [comment]
Hi @dims 
We have finished the items listed above.
Please see the following link as reference:
https://k8s-testgrid.appspot.com/conformance-arm
https://k8s-testgrid.appspot.com/sig-node-arm64

So, can I submit a PR into k8s/community to declare support arm64 ?
Maybe here:
https://github.com/kubernetes/community/blob/a8944bea54d4bb854bdd5da98ecae3ebc7a18d9a/contributors/design-proposals/multi-platform.md

Thanks.

**Comment by BenTheElder** [comment]
Personally I would prefer to see signal for each release monitored by @kubernetes/sig-release before calling something "supported".

https://testgrid.k8s.io/sig-release-1.18-blocking etc.

**Comment by lubinsz** [comment]
Hi @BenTheElder 
Got it.
Thanks for your help.

**Comment by BenTheElder** [comment]
please note that this is just where I think the project should go with this, I've been a bit involved but I do **not** own this (and I'm not actually sure who does?)

**Comment by BenTheElder** [comment]
If SIG Testing does own this, then as a chair I am -1 until @kubernetes/ci-signal @kubernetes/sig-release is monitoring actionable signal to ensure release validty across all releases.
I think SIG release nominally owns this, with some input from e.g. SIG Node.
/sig node
/sig release
CPU builds upstream is coming up in SIG Architecture at the next meeting hopefully, though again I think SIG Release should own this, with _input_ from testing, node, etc.

**Comment by tpepper** [comment]
> Personally I would prefer to see signal for each release monitored by @kubernetes/sig-release before calling something "supported".

I agree.  "Supported" is a strong word.  In the broadest sense it depends on some subset of the contributor base in SIG(s) deciding that intent, then encoding it in tests, which are running successfully and consistently, which then SIG Release accepts as a release blocking test (eg: as per https://github.com/kubernetes/sig-release/blob/master/release-blocking-jobs.md, which itself doesn't define "support").

For hardware architecture and OS support choices I see SIG Node as the driving owner, with many other SIGs/WGs (eg: Cloud Provider, Release, Test, Infra, Docs, ...) needing to be involved too.

In my view SIG Release is largely a trailing delivery vehicle for the project's earlier choices.

**Comment by tinatsou** [comment]
Got you @lubinszARM

**Comment by lubinszARM** [comment]
@tpepper @BenTheElder @dims 
Hi, 
What do you think of the word ‘initial support’?
Can I submit a PR into k8s/community to declare initial support arm64 ?
Thanks.

**Comment by dims** [comment]
@lubinsz we don't want to introduce new terminology as we don't have a written description of what it means. sorry. Looks like the steps at this point are as follows:

0) how do we answer the question - Can anyone outside of your engineering org be able to replicate the conformance tests? (do you need a patch to golang or docker/containerd/runc or any local modifications that is not available publicly)
1) increase the frequency of the CI jobs, less than once a day is way too less for sig-release CI signal team. Please also capture upload more logs similar to existing jobs so someone can diagnose problems independently. Ensure that logs are uploaded even when the run fails, so we can capture flaky tests etc.
2) add an item into sig-node agenda and talk about officially supporting a new architecture
3) add an item into sig-release agenda to walk through adding the CI jobs to release informing board. Once it gets into informing and the CI signal team is happy (1-2 cycles), then it move it into release-blocking.

cc @jdumars

**Comment by neolit123** [comment]
maybe one day each k8s release would enumerate the list of architectures / OSes that are supported:
- what targets the build-system supports.
- artifacts for which ones are shipped on release.
- which ones are e2e tested either by the k8s project or via federated signal and what is the signal health.
- where is help wanted.

**Comment by gianarb** [comment]
Hello!! I had a look at this discussion and the sig-release is in the process of assigning old issues that does not have a sig yet. Related to this work there is this issue https://github.com/kubernetes/kubernetes/issues/87876 from @lubinsz that we do not know who can pick it up. Should we do as @BenTheElder and close it because out of scope for upstream? Thanks

**Comment by lubinsz** [comment]
> Hello!! I had a look at this discussion and the sig-release is in the process of assigning old issues that does not have a sig yet. Related to this work there is this issue #87876 from @lubinsz that we do not know who can pick it up. Should we do as @BenTheElder and close it because out of scope for upstream? Thanks

I have closed it. Thanks.

**Comment by lubinsz** [comment]
Hi @dims 
> 1. how do we answer the question - Can anyone outside of your engineering org be able to replicate the conformance tests? (do you need a patch to golang or docker/containerd/runc or any local modifications that is not available publicly)

I just submitted a prowjob PR with a third-party develooper @hakman 
Please see my PR as reference:
https://github.com/kubernetes/test-infra/pull/18135
@hakman  also submitted some similar test jobs on Arm64:
https://github.com/kubernetes/test-infra/pull/18139
https://github.com/kubernetes/test-infra/pull/18034
In these tests, we didin't use any local modifications 

> 2. increase the frequency of the CI jobs, less than once a day is way too less for sig-release CI signal team. Please also capture upload more logs similar to existing jobs so someone can diagnose problems independently. Ensure that logs are uploaded even when the run fails, so we can capture flaky tests etc.

No problem.

**Comment by hakman** [comment]
I added a new dashboard that uses the `ci/k8s-master` marker:
https://testgrid.k8s.io/kops-misc#kops-aws-arm64-ci

I am trying to fix the DisruptionController, seems a pretty easy fix:
https://github.com/kubernetes/kubernetes/pull/92745

The storage related tests are blocked by CSI images that are not yet multi-arch.

Not sure what is wrong with `[sig-cli] Kubectl client Simple pod should handle in-cluster config`.
